zcash_client_sqlite/src/lib.rs

```diff
@@ -1262,6 +1285,11 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                         .enumerate()
                     {
                         if let Some(address) = txout.recipient_address() {
+                            // TODO: we really want to only mark outputs when a transaction has been
+                            // *reliably* mined.
+                            #[cfg(feature = "transparent-inputs")]
+                            wallet::mark_ephemeral_address_as_mined(wdb.conn.0, &wdb.params, &address, tx_ref).map_err(SqliteClientError::from)?;
```

Comment:
Because the gap indices change as a result of this operation, there can be sent outputs that are newly discoverable as ephemeral, that we previously categorized as external (which can happen if we have two wallets with the same seed competing, for example). We should recategorize them in that case, in particular removing the ephemeral address (which should not be exposed to the wallet user).